### ISSUE TITLE
Floating control show once but hides forever

### ISSUE THREAD

**Issue by tapiama** [smplayer-dev/smplayer#591]
**Describe the bug**
smplayer floating control (full screen mode) shows up once, and when it hides, it never returns

**To Reproduce**
1. play some video, bring to full screen mode
2. move mouse to activate floating control. it shows and works fine.
3. let floating control auto-hide
4. move mouse again to activate floating control, it doesn't show. even after exiting full screen and return to full screen mode.

**Expected behavior**
floating control shows on mouse movement (e.g. normal, or only at bottom of screen), hides after timeout.  shows again when mouse moves.

**Screenshots**
n/a

**Your Enviroment**
 - SMPlayer version: 22.2.0
 - OS: ubuntu 22.10
 - Multimedia engine you're using with SMPlayer (preferences -> general):
 [ ] MPlayer
 [x] MPV

**Logs**
Reproduce the bug in your computer and attach the smplayer log (menu View) here.

**Additional context**
* the bug only appeared immediately after upgrading to ubuntu 22.10.  it worked fine in previous versions. same change/issue on two upgraded ubuntu laptops (macbook air, asus rog)
* the bug isn't fixed by changing video driver between default, libmpv, gpu, vdpau, xv

**Comment by sleeksorrow** [comment]
Can confirm with Ubuntu 22.10 using KDE Plasma 5.26.3 and latest smplayer 22.7.0.
Can reproduce with empty smplayer configuration.
Tried mpv 0.34.1 from ubuntu repositories as well as 0.35 self compiled.
Tried different video outputs and skins.
Tried with 22.7.0 built from sources.

Floating control bar works just once after smplayer start and only returns after restarting smplayer

**Comment by sleeksorrow** [comment]
Can NOT reproduce under Manjaro running KDE Plasma 5.26.3 and latest smplayer 22.7.0

**Comment by thevladsoft** [comment]
It happens on debian testing, KDE Plasma 5.26.3, smplayer 22.7.

**Comment by DaviMedrade** [comment]
Can reproduce in Ubuntu 22.10 with NVIDIA binary drivers (520) and mpv master (compiled using mpv-build using ffmpeg and libplacebo master) with any vo. But in my case it's not that the floating control doesn't show, it's just that it shows behind the video. You can find out if that's the case for you by playing a video that is narrower than the floating control in full screen, or by rotating a widescreen video 90 degrees (Right-click → Video → Rotate; you might need to go into Preferences → Performance tab and set Hardware Decoding to “None” for video rotation to work).

![image](https://user-images.githubusercontent.com/39162/204720951-3061322d-4006-4b61-b46c-17591a086b56.png)

**Comment by thevladsoft** [comment]
It seems to have solved it self on debian testing after last update, so it could have been a problem on xwindow, mesa or qt (there where updates on all those packages, and I didn't check smaplayer until the updates had ended).

**Comment by sleeksorrow** [comment]
> But in my case it's not that the floating control doesn't show, it's just that it shows behind the video. You can find out if that's the case for you by playing a video that is narrower than the floating control in full screen, or by rotating a widescreen video 90 degrees (Right-click → Video → Rotate; you might need to go into Preferences → Performance tab and set Hardware Decoding to “None” for video rotation to work).

Sorry, cannot reproduce this specific behavior. Maybe still related, tho.

On a sidenote, I can rotate the video even with enabled hardware decoding.
```
[vo/gpu/opengl] GL_VERSION='4.4.0 NVIDIA 525.60.11'
[...]
[vd] Using hardware decoding (vdpau-copy).
```

**Comment by tapiama** [comment]
> 

smplayer on ubuntu 22.10 (with latest unphased updates as of 12-04-2022) still have the same issue, e.g. floating control bar disappear on second activation/mouse event (confirmed as a always-on-top UI element not working, when displaying non-full-width videos in full screen)

perhaps unrelated or related, is that docked playlist failed to scroll at all when entering and than exiting full-screen.  perhaps this is related to the changes in [gnome UI stack](https://gitlab.gnome.org/GNOME/gtk/-/issues/2971) (which also messed up my nautilus file manager scrolling for large lists)

on the side note, the snap version (ubuntu) of smplayer don't have this issue and can be a temp work around (but it has a different problem - displaying non-English characters as boxes)

**Comment by sleeksorrow** [comment]
Flatpak version also is not affected, but quite dated.
https://flathub.org/apps/details/info.smplayer.SMPlayer

**Comment by chevcheli0s** [comment]
I have same issue on Ubuntu Mate 20.10
I can see bottom bar if I change scale by pressing "w"
But that's not the solution.

**Comment by memory-colors** [comment]
Interestingly, this issue will still be there if you apt-mark hold smplayer and do the upgrade from (in my case) Kubuntu 22.04 to 22.10. My Smplayer is showing:

Version: 21.10.0 (revision 10000)
Using Qt 5.15.6 (compiled with Qt 5.15.2)
Using MPV 0.34.1

But both this issue and the issue with the playlist scroll breaking after using fullscreen are present.

I tried this after just doing a straight upgrade borked smplayer. Reinstalled Kubuntu 22.04 and Smplayer, all was well. Put smplayer (and mplayer, mpv, smplayer-l10, smplayer-themes) on hold, did the upgrade, issue is there again. Happy to provide any other info.

**Comment by memory-colors** [comment]
> Can NOT reproduce under Manjaro running KDE Plasma 5.26.3 and latest smplayer 22.7.0

What QT version is Manjaro using?

**Comment by sleeksorrow** [comment]
> What QT version is Manjaro using?

QT is at version 5.15.7 on Manjaro right now.

**Comment by memory-colors** [comment]
Ok, here's what I've got:

Operating System: Kubuntu 22.10
KDE Plasma Version: 5.26.4
KDE Frameworks Version: 5.100.0
Qt Version: 5.15.6
Kernel Version: 5.19.0-27-generic (64-bit)
Graphics Platform: X11
Processors: 8 × AMD Ryzen 5 3550H with Radeon Vega Mobile Gfx
Memory: 13.6 GiB of RAM
Graphics Processor: AMD Radeon Vega 8 Graphics

Issue is showing here, with Smplayer 21.10.0 and Smplayer 22.7.0

***BUT***

I just tried the Flatpak, which is 22.2 and it does *not* have this issue.

**Comment by sleeksorrow** [comment]
Okay, so after thinking about it: smplayer itself is not the cause, or else every distribution would be affected. But on Ubuntu it seems every hardware combination is affected. My system looks very different to yours with older Intel CPU and Nvidia binary drivers.

So it's some library from the system that is at fault, and the flatpak version uses it's own bundled unaffected version.

So I took a look into my flatpak installation in `~/.local/share/flatpak/app/info.smplayer.SMPlayer/x86_64/stable/active/files/lib` (I installed in user space) and the only libraries there that in my opinion have a clear connection to UI is Qt5WebKit and Qt5WebKitWidget. So I strongly suspect the QT libraries shipping with Ubuntu to be the cause for the bug in non-flatpak/snap versions.

That said, I'm not really firm with flatpaks sandbox technique and I'm just guessing.

**Comment by DaviMedrade** [comment]
> I can rotate the video even with enabled hardware decoding.

I tested now and I can rotate the video with any of the `-copy` decoders, (~~as well as with `vdpau`~~ edit: for some reason `vdpau` doesn't work in my system so it reverts back to software decoding), but I can't with `cuda` nor with “Auto”, which, in my machine, for H.264, auto-selects `nvdec`. By the way, `nvdec` is not shown as an option in the hardware decoders list in SMPlayer even though it is supported by mpv (and when I run `mpv --hwdec=help` to show the list of supported hardware decoders `nvdec` is in the list).

Edit: created #610 (re: no `nvdec` in the hardware decoders list).

**Comment by memory-colors** [comment]
> Okay, so after thinking about it: smplayer itself is not the cause, or else every distribution would be affected. But on Ubuntu it seems every hardware combination is affected. My system looks very different to yours with older Intel CPU and Nvidia binary drivers.
> 
> So it's some library from the system that is at fault, and the flatpak version uses it's own bundled unaffected version.
> 
> So I took a look into my flatpak installation in `~/.local/share/flatpak/app/info.smplayer.SMPlayer/x86_64/stable/active/files/lib` (I installed in user space) and the only libraries there that in my opinion have a clear connection to UI is Qt5WebKit and Qt5WebKitWidget. So I strongly suspect the QT libraries shipping with Ubuntu to be the cause for the bug in non-flatpak/snap versions.
> 
> That said, I'm not really firm with flatpaks sandbox technique and I'm just guessing.

I don't follow your first statement. It doesn't seem hardware-dependant at all? It seems like whatever updates happened in 'buntu (and I think someone mentioned Mint?) cause the issue on Smplayer. But I'm also in the dark, just hoping it gets fixed. My favorite media player!

**Comment by sleeksorrow** [comment]
>It doesn't seem hardware-dependant at all?

Maybe I phrased it in a confusing way, but that's what I meant. It's not hardware, not smplayer, not X, not graphics drivers. It's only something Ubuntu does specifically. And my guess is that it's something about how they build QT for Ubuntu.

I'm also on my limits of making useful educated guesses and I will now stop guessing further, this leads to nowhere. From here we need @smplayer-dev. Or enough time to finally get rid of Ubuntu on our machines....

**Comment by memory-colors** [comment]
> > It doesn't seem hardware-dependant at all?
> 
> Maybe I phrased it in a confusing way, but that's what I meant. It's not hardware, not smplayer, not X, not graphics drivers. It's only something Ubuntu does specifically. And my guess is that it's something about how they build QT for Ubuntu.
> 
> I'm also on my limits of making useful educated guesses and I will now stop guessing further, this leads to nowhere. From here we need @smplayer-dev. Or enough time to finally get rid of Ubuntu on our machines....

Gotcha. Yeah I would guess QT, but also I'm pretty far from an expert. 

Whatever it is, it seems definitely related to the playlist bug.

**Comment by smplayer-dev** [comment]
Is it only happening on Ubuntu 22.10?

**Comment by sleeksorrow** [comment]
Our sample size is very small with only ~6 hosts. But so far we have Ubuntu 22.10, Mate 20.10 and temporarily Debian Testing which seemingly healed with some update.

**Comment by memory-colors** [comment]
> Is it only happening on Ubuntu 22.10?

I'm on Kubuntu 22.10. See my reply above, this happens even if Smplayer is put on hold in Kubuntu 22.04, then upgraded to 22.10. Thanks for your work, btw!!

**Comment by DaviMedrade** [comment]
Since it seems to be related to Qt, my interpretation of the problem (video appearing in front of the floating control) happens in Ubuntu 22.10. According to SMPlayer: Using Qt 5.15.6 (compiled with Qt 5.15.3).

**Comment by smplayer-dev** [comment]
I installed Ubuntu 22.10 on a virtual machine. It seems the problem happens when using xorg, but not when using xwayland.

**Comment by sleeksorrow** [comment]
Darn, Wayland on Nvidia proprietary driver is still a bloody mess for me, tried a few times but the result never was bug free enough for every day use. But nevertheless, your finding is a strong pointer to some Ubuntu Xorg packaging as the root cause. Thanks!

**Comment by smplayer-dev** [comment]
Maybe I was wrong yesterday. I've just tested smplayer 22.7 from the ppa and it works fine in xorg, but the floating control gets partially hidden in wayland.

**Comment by sleeksorrow** [comment]
Well this is surprising indeed, since this would be the first machine here in this issue report with Ubuntu 22.10 Xorg that does NOT show this bug. I wonder if we can find the difference.

**Comment by BileDemon** [comment]
This won't help much, but the SMPlayer on my system is plagued by this same bug for years now, beginning with Ubuntu Mate 20.04 LTS, and now also on the latest Kubuntu 22.10 using SMPlayer 22.7.0. In my case it's not Nvidia but an AMD system. And I also noticed that with narrower videos, the floating control is hidden behind the video.

I really hope this problem can be fixed any time soon, since I prefer SMPlayer over VLC.

**Comment by smplayer-dev** [comment]
It's difficult to find a solution if the problem is actually (it seems) in another library. 
A way to fix it would be to get rid of the floating control and use the normal control, which would be shown/hidden when the mouse is moved, but it has the side effect that the video area is resized. This is how early versions of smplayer did it, and people didn't like it.

**Comment by cipricus** [comment]
> cannot reproduce this specific behavior.

Have you tried different aspect ratios (pressing A)? With none enabled there is no floating toolbar, but it is visible partially, totally or covered by the video depending on the aspect ratio.

**Comment by cipricus** [comment]
Snap and appimage versions are not affected. 

(But on my Kubuntu 22.10 the appimage needed fixing like [here](https://github.com/project-slippi/Ishiiruka/issues/323#issuecomment-977415376).)

**Comment by DaviMedrade** [comment]
I don't see anyone else reporting this specifically stating that they use mplayer, so I thought it might be some peculiar interaction between mpv and smplayer when smplayer runs with the Ubuntu-provided Qt libs, but it seems not to be the case, since the same behavior happens when using mplayer: when I start the video and go to fullscreen, the floating control appears normally, but after it hides and I move the mouse to make it reappear, it appears behind the video, like in the image below.

![image](https://user-images.githubusercontent.com/39162/216367729-4d2366a4-0661-432c-9c32-5377902ab9e7.png)

**Comment by DaviMedrade** [comment]
Both the mplayer/mpv log window, as well as Guake (a floating terminal, unrelated to smplayer) are able to appear on top of the video. So the bug is not that Qt somehow makes it impossible for a window to appear on top of the video (I thought that could be the case due to some hardware acceleration aspect, but it wouldn't be the case anyway because it happens even when the player usees video output drivers that don't use hardware acceleration).

![image](https://user-images.githubusercontent.com/39162/216371271-8095d93a-bb3d-4bf0-8e25-450a71719b3c.png)

**Comment by cipricus** [comment]
@DaviMedrade - Why isn't that happening in appimage and snap versions?

**Comment by DaviMedrade** [comment]
I don't know a lot about appimage and snap packages, but, based on what little I know, my guess would be that it's because those kinds of packages bring their own libs (including the Qt libs) instead of using the ones provided by the OS.

**Comment by DaviMedrade** [comment]
This problem seems to have gone away once I updated to Ubuntu 23.04.

The About screen shows:
Version: 22.7.0 (revision 10091)
Using Qt 5.15.8 (compiled with Qt 5.15.6)

**Comment by BileDemon** [comment]
Yes, I can confirm that the problem seems to have resolved itself by upgrading to Kubuntu 23.04.
Thank god, I can finally use SMPlayer properly and forget about this.

**Comment by sleeksorrow** [comment]
Resolved for me as well with Ubuntu 23.04.

**Comment by tapiama** [comment]
Me too, confirmed Ubuntu 23.04 issue seem resolved! Thanks to whoever made it work!!
